symbiflow.py

```diff
@@ -401,14 +405,19 @@ def get_overhead_runtime(logfile):
         fasm_log = os.path.join(self.out_dir, 'fasm.log')
 
         log['pack'] = get_step_runtime('Packing', pack_log, 3)
-        log['overhead'] = get_overhead_runtime(pack_log)
         log['place'] = get_step_runtime('Placement', place_log, 3)
-        log['overhead'] += get_overhead_runtime(place_log)
         log['route'] = get_step_runtime('Routing', route_log, 3)
-        log['overhead'] += get_overhead_runtime(route_log)
         # XXX: Need add to genfasm the amount of time it took to create the fasm file.
         #      For now the whole command execution time is considered
         log['fasm'] = get_step_runtime('The entire flow of VPR', fasm_log, 6)
+        if "pack_all" in self.unprinted_runtimes and "place_all" in self.unprinted_runtimes and "route_all" in self.unprinted_runtimes and "fasm_all" in self.unprinted_runtimes:
+            log['overhead'] = self.unprinted_runtimes['pack_all'] - log['pack']
+            log['overhead'
+                ] += self.unprinted_runtimes['place_all'] - log['place']
+            log['overhead'
+                ] += self.unprinted_runtimes['route_all'] - log['route']
+            log['overhead'
+                ] += self.unprinted_runtimes['fasm_all'] - log['fasm']
```

Comment:
I would leave fasm out of these calculations, as it is required for us to modify VPR to get the useful work of the genfasm step, which is not reported in the fasm.log.

Why is the if statement needed? maybe would be worth to use the function to get the overhead time that checks if a step is in the unprinted_runtimes and, if it is, returns `pack_all - pack`

Reply:
I removed fasm from calculations and moved this code to function.

I didn't checked exactly why this if statement is needed, but generally it is failing when `add_runtimes` is called. This function is shared between `nextpnr` and  `vpr`, so I think it is failing, when `nextpnr` toolchain is selected as it do not have `pack_all` runtime.